homeassistant/components/soma/__init__.py

```diff
@@ -105,6 +107,12 @@ def device_info(self):
             "manufacturer": "Wazombi Labs",
         }
 
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        attrs = {ATTR_BATTERY_LEVEL: self.battery_level}
```

Comment:
We prefer to create separate sensor entities to report battery level. A device can be represented by many different entities.